tests/Console/SelfUpdate/NewVersionCheckerTest.php

```diff
@@ -0,0 +1,361 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Console\SelfUpdate;
+
+use PhpCsFixer\Console\SelfUpdate\NewVersionChecker;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Console\SelfUpdate\NewVersionChecker
+ */
+final class NewVersionCheckerTest extends TestCase
+{
+    /**
+     * @param string      $currentVersion
+     * @param null|string $expectedVersion
+     *
+     * @dataProvider getLatestVersionCases
+     */
+    public function testGetLatestVersion($currentVersion, $expectedVersion)
+    {
+        $checker = new NewVersionChecker($currentVersion, $this->createGithubClientStub());
+
+        $this->assertSame($expectedVersion, $checker->getLatestVersion());
+    }
+
+    public function getLatestVersionCases()
+    {
+        return array(
+            array('v1.0.0', 'v2.4.1'),
+            array('v1.2.0', 'v2.4.1'),
+            array('v1.2.5', 'v2.4.1'),
+            array('v2.0.0', 'v2.4.1'),
+            array('v2.0.0', 'v2.4.1'),
+            array('v2.0.0', 'v2.4.1'),
+            array('v2.0.0', 'v2.4.1'),
+            array('v2.2.0', 'v2.4.1'),
+            array('v2.2.5', 'v2.4.1'),
+            array('v2.4.0', 'v2.4.1'),
+            array('v2.4.1', null),
+            array('v2.4.2', null),
+            array('v2.5.0', null),
+            array('v3.0.0', null),
+            array('v3.2.0', null),
+            array('v3.2.5', null),
+        );
+    }
+
+    /**
+     * @param string      $currentVersion
+     * @param null|string $expectedVersion
+     *
+     * @dataProvider getLatestVersionOfCurrentMajorCases
+     */
+    public function testGetLatestVersionOfCurrentMajor($currentVersion, $expectedVersion)
+    {
+        $checker = new NewVersionChecker($currentVersion, $this->createGithubClientStub());
+
+        $this->assertSame($expectedVersion, $checker->getLatestVersionOfCurrentMajor());
+    }
+
+    public function getLatestVersionOfCurrentMajorCases()
+    {
+        return array(
+            array('v1.0.0', 'v1.13.2'),
+            array('v1.2.0', 'v1.13.2'),
+            array('v1.2.5', 'v1.13.2'),
+            array('v2.0.0', 'v2.4.1'),
+            array('v2.2.0', 'v2.4.1'),
+            array('v2.2.5', 'v2.4.1'),
+            array('v2.4.0', 'v2.4.1'),
+            array('v2.4.1', null),
+            array('v2.5.0', null),
+            array('v3.0.0', null),
+            array('v3.2.0', null),
+            array('v3.2.5', null),
+        );
+    }
+
+    private function createGithubClientStub()
+    {
+        $githubClient = $this->prophesize('PhpCsFixer\Console\SelfUpdate\GithubClient');
```

Comment:
you shall prophesize only interfaces.
why this internal class is not final? :P

Reply:
Interface added.